discussion/app/views/fragments/commentShare.scala.html

```diff
@@ -10,29 +10,27 @@
     <div class="sharing-buttons" data-link-name="comment social">
     @defining(s"${comment.webUrl}") { permalink =>
         @defining(StringEscapeUtils.unescapeHtml(Jsoup.clean(comment.body.replaceAll("<blockquote>.*</blockquote>", ""), Whitelist.none()))) { commentBody =>
-            <a href="@ShareLinks.createShareLinkForComment(platform = Facebook, href = permalink, text = "", quote = Some(s"""${comment.profile.displayName} commented: "$commentBody"""")).href"
-            target="_blank"
-            class="social__action social-icon-wrapper"
-            data-link-name="social-comment : facebook">
-                <span class="inline-icon__fallback button">
-                    Share on Facebook</span>
-                @fragments.inlineSvg("share-facebook", "icon", List("rounded-icon", "social-icon", "social-icon--facebook", "comment-facebook-icon"))
-                <span class="u-h">Facebook</span>
-            </a>
+            @shareLink(permalink, Facebook, "", Some(s"""${comment.profile.displayName} commented: "$commentBody""""))
 
             @* Twitter allows 140 characters. We need 2 for the quotes and 24 for the URL. *@
             @defining(if(commentBody.length <= 114) s""""$commentBody"""" else s""""${commentBody.take(111)}..."""") { commentText =>
-                <a href="@ShareLinks.createShareLinkForComment(platform = Twitter, href = permalink, text = commentText).href"
-                target="_blank"
-                class="social__action social-icon-wrapper"
-                data-link-name="social-comment : twitter">
-                    <span class="inline-icon__fallback button">
-                        Share on Facebook</span>
-                    @fragments.inlineSvg("share-twitter", "icon", List("rounded-icon", "social-icon", "social-icon--twitter", "comment-twitter-icon"))
-                    <span class="u-h">Twitter</span>
-                </a>
+                @shareLink(permalink, Twitter, commentText, None)
             }
         }
     }
+
+    @shareLink(permalink: String, platform: SharePlatform, text: String, quote: Option[String]) = {
```

Comment:
I tried making this `@shareLink(permalink: String)(platform: ....)`, but the templating didn't seem to like defining `@shareLinkArticle = { @shareLink(permalink) }` when the permalink was in scope